Assets/Scripts/SS3D/Systems/Chat/ChatWindow.cs

```diff
@@ -0,0 +1,179 @@
+﻿using FishNet;
+using SS3D.Core;
+using SS3D.Core.Behaviours;
+using SS3D.Systems.Entities;
+using SS3D.Systems.Inputs;
+using SS3D.Systems.PlayerControl;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using TMPro;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+using InputSystem = SS3D.Systems.Inputs.InputSystem;
+
+namespace SS3D.Engine.Chat
+{
+    /// <summary>
+    /// Behaviour responsible for handling chat functionality.
+    /// </summary>
+    public abstract class ChatWindow : View, IPointerEnterHandler, IPointerExitHandler
+    {
+        [SerializeField] protected ChatChannels chatChannels = null;
+        [SerializeField] protected TMP_InputField inputField = null;
+        [SerializeField] private TextMeshProUGUI chatText = null;
+        
+        [HideInInspector] public List<string> availableChannels = new List<string>();
+        
+        private readonly List<ChatMessage> _chatMessages = new List<ChatMessage>();
+        
+        private InputSystem _inputSystem;
+        private Controls.OtherActions _controls;
+        private bool _initialized;
+
+        protected virtual ChatChannel GetCurrentChatChannel() => null;
+
+        protected override void OnAwake()
+        {
+            base.OnAwake();
+            
+            _inputSystem = Subsystems.Get<InputSystem>();
+            _controls = _inputSystem.Inputs.Other;
+            ChatMessageSender.InitializeIfNeeded();
```

Comment:
Here's your issue. You need to initialize a manager from a UI, this makes no sense. By making the change to a system you'll avoid this need, reducing code nesting